src/python/tests/test_orthology_benchmark.py

```diff
@@ -0,0 +1,234 @@
+
+
+@pytest.mark.parametrize(
+    "core_names, exp_output, expectation",
+    [
+        (["mus_musculus_core_105_1", "mus_musculus_core_52_105_3", "mus_musculus_core_104_4"],
+         "mus_musculus_core_52_105_3", does_not_raise()),
+        ([], None, raises(ValueError,
+                          match=r"Empty list of core databases. Cannot determine the latest one."))
+    ]
+)
+def test_find_latest_core(core_names: List[str], exp_output: str, expectation: ContextManager) -> None:
+    """Tests :func:`orthology_benchmark.find_latest_core()` function.
+
+    Args:
+        core_names: A list of core database names.
+        exp_output: Expected return value of the function.
+        expectation: Context manager for the expected exception, i.e. the test will only pass if that
+            exception is raised. Use :class:`~contextlib.nullcontext` if no exception is expected.
+
+    """
+    with expectation:
+        assert orthology_benchmark.find_latest_core(core_names) == exp_output
+
+
+@pytest.mark.parametrize(
+    "multi_dbs",
+    [
+        [{'src': 'core/danio_rerio_core_105_11', 'name': 'danio_rerio_core_105_11'},
+         {'src': 'core/mus_musculus_cbaj_core_107_1', 'name': 'mus_musculus_cbaj_core_107_1'},
+         {'src': 'core/mus_musculus_core_106_39', 'name': 'mus_musculus_core_106_39'}]
+    ],
+    indirect=True
+)
+class TestGetCoreNames:
+    """Tests :func:`orthology_benchmark.get_core_names()` function.
+
+    Attributes:
+        core_dbs: A set of test core databases.
+
+    """
+
+    core_dbs = {}  # type: Dict
+    host = None  # type: str
+    port = None  # type: int
+    username = None  # type: str
+
+    # autouse=True makes this fixture be executed before any test_* method of this class, and scope='class' to
+    # execute it only once per class parametrization
+    @pytest.fixture(scope='class', autouse=True)
+    def setup(self, request: FixtureRequest, multi_dbs: Dict) -> None:
+        """Loads the required fixtures and values as class attributes.
+
+        Args:
+            request: Access to the requesting test context.
+            multi_dbs: Dictionary of unit test databases (fixture).
+
+        """
+        type(self).core_dbs = multi_dbs
+        server_url = sqlalchemy.engine.url.make_url(request.config.getoption('server'))
+        type(self).host = server_url.host
+        type(self).port = server_url.port
+        type(self).username = "ensro" if server_url.username == "ensadmin" else server_url.username
+
+    @pytest.mark.parametrize(
+        "species_names, exp_output, expectation",
+        [
+            (["danio_rerio", "mus_musculus", "zea_mays"],
+             {"danio_rerio": os.environ['USER'] + "_danio_rerio_core_105_11",
+              "mus_musculus": os.environ['USER'] + "_mus_musculus_core_106_39"},
+             does_not_raise()),
+            ([], None, raises(ValueError,
+                              match=r"Empty list of species names. Cannot search for core databases."))
```

Comment:
```suggestion
            (
                ["danio_rerio", "mus_musculus", "zea_mays"],
                {"danio_rerio": os.environ['USER'] + "_danio_rerio_core_105_11",
                 "mus_musculus": os.environ['USER'] + "_mus_musculus_core_106_39"},
                does_not_raise()
            ),
            ([], None, raises(ValueError,
                              match=r"Empty list of species names. Cannot search for core databases."))
```

Reply:
Same as before.

Reply:
Aha, this one wasn't by mistake. Basically, I'm/was wondering why `(` and `)` should be singletons in lines 206 and 211 - this is perhaps the only such case in our code base. In all my `pytest` scripts, for none of test cases they are in their own lines.

Reply:
It is a matter of block readability: if you don't pay attention, one could mix like 210 as part of the block started in line 206, so "sacrificing" two lines for that seems as reasonable as when you put empty lines within the code in your functions.